File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        stack<char>st;
        for(char ch : s) {
            if(ch == ']' && st.empty()) continue;
            else if(ch == ']' && st.top() == '[') st.pop();
            else st.push('[');
        }
        return (st.size()+1)/2;
    }
};

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        int n = s.length();
        int res = 0;
        for(int i=0;i<n;i++) {
            char ch = s.charAt(i);
            if(ch == '[') res++;
            else if(res > 0) res--;
        }
        return (res+1)/2;
       
    }
}